### PR TITLE
Move lmnet/executor/predict.py to blueoil/cmd

### PR DESCRIPTION
**Pull request by tfujiwar**
## What this patch does to fix the issue.
- Remove blueoil/cmd/predict.py (3916fac)
- Move lmnet/executor/predict.py to blueoil/cmd (fbe78d5)
- Update blueoil/cmd/predict.py (d1a48fb)

This PR looks big but most of the changes are just copied. Maybe it's easier to check each commit one by one than checking the whole changes.

## Link to any relevant issues or pull requests.
#746

### CODE REVIEW

File: blueoil/cmd/predict.py [hunk 1]
```diff
@@ -13,21 +13,160 @@
...
+    writer = OutputWriter(
+        task=config.TASK,
+        classes=config.CLASSES,
+        image_size=config.IMAGE_SIZE,
+        data_format=config.DATA_FORMAT
+    )
+
+    results = []
+    for step in range(step_size):
+        start_index = (step) * config.BATCH_SIZE
```
Comment (tsawada) [inline blueoil/cmd/predict.py]: remove parenthesis

File: blueoil/cmd/predict.py [hunk 2]
```diff
@@ -13,21 +13,160 @@
...
+        end_index = (step + 1) * config.BATCH_SIZE
+
+        image_files = all_image_files[start_index:end_index]
+
+        while len(image_files) != config.BATCH_SIZE:
```
Comment (tsawada) [inline blueoil/cmd/predict.py]: this is inefficient, and will loop forever if len(image_files) > config.BATCH_SIZE.
```
if len > size:
    files += [DUMMY] * (len - size)
```

File: blueoil/cmd/predict.py [hunk 3]
```diff
@@ -13,21 +13,160 @@
...
+            # add dummy image.
+            image_files.append(DUMMY_FILENAME)
+
+        images, raw_images = _get_images(
+            image_files, config.DATASET.PRE_PROCESSOR, config.DATA_FORMAT)
+
+        feed_dict = {images_placeholder: images}
+        outputs = sess.run(output_op, feed_dict=feed_dict)
```
Comment (tsawada) [inline blueoil/cmd/predict.py]: put feed_dict in a single line

File: blueoil/cmd/predict.py [hunk 4]
```diff
@@ -13,21 +13,160 @@
...
+
+        if config.POST_PROCESSOR:
+            outputs = config.POST_PROCESSOR(outputs=outputs)["outputs"]
+
+        results.append(outputs)
+
+        writer.write(
```
Comment (tsawada) [inline blueoil/cmd/predict.py]: can you concatenate into fewer lines?

File: blueoil/cmd/predict.py [hunk 5]
```diff
@@ -13,21 +13,160 @@
...
+
+
+def run(input_dir, output_dir, experiment_id, config_file, restore_path, save_images):
+    environment.init(experiment_id)
+    config = config_util.load_from_experiment()
+    if config_file:
+        config = config_util.merge(config, config_util.load(config_file))
+
+    if not os.path.isdir(input_dir):
+        raise Exception("Input directory {} does not exist.".format(input_dir))
```
Comment (tsawada) [inline blueoil/cmd/predict.py]: Avoid raising `Exception` and use more specific ones. In this case,  `FileNotFoundError` might be a good choice

**Comment by tfujiwar** [comment]
@tsawada Thank you for the comments! Let me clean up `train.py` and `predict.py` in other PRs.

**Pull request merged**